cli/medperf/tests/commands/test_view.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import yaml
+import json
+from medperf.entities.interface import Entity
+from medperf.exceptions import InvalidArgumentError
+from medperf.commands.view import EntityView
+
+
+def expected_output(entities, format):
+    if isinstance(entities, list):
+        data = [entity.todict() for entity in entities]
+    else:
+        data = entities.todict()
+
+    if format == "yaml":
+        return yaml.dump(data)
+    if format == "json":
+        return json.dumps(data)
+
+
+def generate_entity(id, mocker):
+    entity = mocker.create_autospec(spec=Entity)
+    mocker.patch.object(entity, "todict", return_value={"id": id})
+    return entity
+
+
+@pytest.fixture
+def ui_spy(mocker, ui):
+    return mocker.patch.object(ui, "print")
+
+
+@pytest.fixture(
+    params=[{"local": ["1", "2", "3"], "remote": ["4", "5", "6"], "user": ["4"]}]
+)
+def setup(request, mocker):
+    local_ids = request.param.get("local", [])
+    remote_ids = request.param.get("remote", [])
+    user_ids = request.param.get("user", [])
+    all_ids = list(set(local_ids + remote_ids + user_ids))
+
+    local_entities = [generate_entity(id, mocker) for id in local_ids]
+    remote_entities = [generate_entity(id, mocker) for id in remote_ids]
+    user_entities = [generate_entity(id, mocker) for id in user_ids]
+    all_entities = list(set(local_entities + remote_entities + user_entities))
+
```

Comment:
Let's talk.
I think what we are doing here is "mimicing how entities behave", which we shouldn't do.
Making sure that entity.all is called with the correct mine_only and local_only arguments should be enough for testing this specfic idea. We don't have to trace that all the way to the ui output. what do you think?
I think the entity.list PR illustrates more what I am saying

Reply:
I want to mention btw that in the priority PR I did, I also mimicked how the server behaves, which should be changed.